packages/epics/src/comm.ts

```
@@ -57,7 +57,10 @@ export const commListenEpic = (
                 action.payload.kernelRef === kernelRef
             )
           )
-        )
+        ),
+        catchError((error: Error) => {
+          return empty();
```

Here and below, let's return an error action instead of an empty Observable.

```js
return of(
    actions.executeFailed({
        error: new Error(
        "The WebSocket connection has unexpectedly disconnected."
        ),
        code: errors.EXEC_WEBSOCKET_ERROR,
        contentRef
    })
);
```

---

I update two locations. Is contentRef available in watchExecutionStateEpic for KillKernelSuccessful action?

---

Yes, it should be accessible under the `action.payload.contentRef` property of the `NewKernelAction`.

---

Updated